### PR TITLE
Fixes VS2022 error C5233: explicit lambda capture 'isSlash' is not used

### PR DESCRIPTION
**Pull request by amzn-jillich**
Signed-off-by: Benjamin Jillich <jillich@amazon.com>

### CODE REVIEW

File: Code/Tools/AssetProcessor/native/FileWatcher/FileWatcher.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ static bool IsSubfolder(const QString& folderA, const QString& folderB)
     using AZStd::begin;
     using AZStd::end;
 
-    constexpr auto isSlash = [](const QChar c) constexpr
+    auto isSlash = [](const QChar c) constexpr
```
Comment (jeremyong) [inline Code/Tools/AssetProcessor/native/FileWatcher/FileWatcher.cpp:28]: Should we not just remove `isSlash` from the capture groups in the lambdas below? Static storage is automatically captured in lambdas as I understand it.

Comment (Bindless-Chicken) [reply Code/Tools/AssetProcessor/native/FileWatcher/FileWatcher.cpp:28]: I would also recommend if that doesn't work to at least keep the `const`.

Comment (amzn-jillich) [reply Code/Tools/AssetProcessor/native/FileWatcher/FileWatcher.cpp:28]: That actually makes a lot of sense! Changed

Comment (amzn-jillich) [reply Code/Tools/AssetProcessor/native/FileWatcher/FileWatcher.cpp]: Had to reverse, while VS2022 handles it the way you say, VS2019 doesn't like that approach.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by amznestebanpapp — APPROVED** [review]
I have seen this false-positive before, the constexpr lambdas dont need to be passed, however, some versions in 2019 need it.
I am also ok if you want to leave the `constexpr` and mark it with `[[maybe_unused]]`

**Pull request merged**